aten/src/ATen/core/op_registration/kernel_functor.h

```diff
@@ -106,66 +123,111 @@ namespace detail {
+  template<class Key, class Value>
+  struct return_type_to_ivalue<c10::Dict<Key, Value>> {
+    static IValue call(c10::Dict<Key, Value>&& v) {
+      static_assert(guts::typelist::contains<supported_primitive_arg_types, Key>::value, "You tried to register a kernel with an unsupported return type: Dict<Key, Value> and Key is not one of the supported primitive types.");
+      static_assert(guts::typelist::contains<supported_primitive_arg_types, Value>::value, "You tried to register a kernel with an unsupported return type: Dict<Key, Value> and Value is not one of the supported primitive types.");
+      return IValue(impl::toGenericDict(std::move(v)));
+    }
+  };
+  // The following specialisations of return_type_to_ivalue are technically not
   // necessary since we would hit the base case and show an error message
   // there if they didn't exist, but we can show a better error message
   // in some common error scenarios.
   template<class T>
-  struct return_type_to_ivalue_<c10::ArrayRef<T>> {
+  struct return_type_to_ivalue<c10::ArrayRef<T>> {
     static_assert(guts::false_t<c10::ArrayRef<T>>::value, "You tried to register a kernel with an unsupported return type: c10::ArrayRef<T>. Please use std::vector<T> instead.");
   };
+  template<class Key, class Value>
+  struct return_type_to_ivalue<std::unordered_map<Key, Value>> {
+    static_assert(guts::false_t<std::unordered_map<Key, Value>>::value, "You tried to register a kernel with an unsupported return type: std::unordered_map<Key, Value>. Please use c10::Dict<Key, Value> instead.");
+  };
   template<class T>
-  struct return_type_to_ivalue_<T, guts::enable_if_t<std::is_same<float, T>::value>> {
+  struct return_type_to_ivalue<T, guts::enable_if_t<std::is_same<float, T>::value>> {
     static_assert(guts::false_t<T>::value, "You tried to register a kernel with an unsupported return type: float. Please use double instead.");
   };
   template<class T>
-  struct return_type_to_ivalue_<T, guts::enable_if_t<std::is_same<const char*, T>::value>> {
+  struct return_type_to_ivalue<T, guts::enable_if_t<std::is_same<const char*, T>::value>> {
     static_assert(guts::false_t<T>::value, "You tried to register a kernel with an unsupported return type: const char*. Please use std::string instead.");
   };
   template<class T>
-  struct return_type_to_ivalue_<T, guts::enable_if_t<std::is_integral<T>::value && !guts::typelist::contains<supported_primitive_arg_types, T>::value>> {
+  struct return_type_to_ivalue<T, guts::enable_if_t<std::is_integral<T>::value && !guts::typelist::contains<supported_primitive_arg_types, T>::value>> {
     static_assert(guts::false_t<T>::value, "You tried to register a kernel with an unsupported integral return argument type. Please use int64_t instead.");
   };
-
+  // legacy_return_type_to_ivalue is like return_type_to_ivalue but additionally
+  // allows a few deprecated types like std::vector.
```

Comment:
Did you mean to have vector here as well?

Reply:
ah no, vector is actually still allowed, possibly to be deprecated in a future diff. The comment should say "unordered_map". Fixed it.